### PR TITLE
Introduce filters for channel, customer country, and risk level on the transactions list page

### LINKED ISSUE #7596: Transactions Report - Add filters for other useful columns in UI
### Description

The **Channel** column on the Transactions page indicates whether a transaction was made online or in person. Currently, there's no way of filtering out transactions of a particular type (channel).

Similarly, we can add filters for **Country** and **Risk Level**.

### Acceptance criteria

* Add a new filter called Channel in the advanced filters UI that will let a merchant show only online or only in-person orders.
* Add a new filter for Country.
* Add a new filter for Risk Level.

### Screenshot

image

### PR DESCRIPTION
**Pull request by mordeth**
#### Changes proposed in this Pull Request
Introducing three additional filters for the transactions list page:
- Channel: Online or In-Person
- Customer Country: All countries worldwide
- Risk Level: Normal, Elevated or Highest

https://github.com/Automattic/woocommerce-payments/assets/8667118/404cccaa-ee9d-4971-8f47-4fdb9beb91d2

#### Testing instructions
**To test the changes, use pull request #4205 on the server side**

**Case 1**
- Navigate to Payment -> Transactions.
- Apply an advanced filter for Channel.
- Optionally include In-Person transactions to validate the proper functionality of the filter.
- Choose 'Online' as the selection criteria.
- Verify that the listed transactions are specifically withOnline channel.
- Confirm that both the transactions summary and pagination are functioning correctly.

**Case 2**
- Navigate to Payment -> Transactions.
- Apply an advanced filter for Channel.
- Create In-Person transactions to validate the proper functionality of the filter.
- Choose 'In-Person' as the selection criteria.
- Verify that the listed transactions are specifically withIn-Person channel.
- Confirm that both the transactions summary and pagination are functioning correctly.

**Case 3**
- Navigate to Payment -> Transactions.
- Apply an advanced filter for Customer Country.
- Select any country with which you generate transactions as the chosen criteria.
- Verify that the listed transactions are specifically with the selected country.
- Confirm that both the transactions summary and pagination are functioning correctly.

Case 4
- Navigate to Payment -> Transactions.
- Apply an advanced filter for Risk Level.
- Ensure that you have a transaction with a higher risk level.
- Choose 'IS NOT' as the rule and 'Normal' as the criteria.
- Verify that the listed transactions have a risk level different from Normal.
- Confirm that both the transactions summary and pagination are functioning correctly.

*

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7754` or branch name `add/7596-additional-transaction-filters` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/7596-additional-transaction-filters) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 84add7ba4511c07dfd1b2c562135a5d89736abc2
- Build time: 2023-12-01 20:35:11 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by dpaun1985 — APPROVED** [review]
✅ LGTM

File: client/deposits/details/test/index.tsx [hunk 1]
```diff
@@ -60,6 +61,13 @@ describe( 'Deposit overview', () => {
 				},
 			},
 		};
+		global.wcSettings = {
```
Comment (anu-rock) [inline client/deposits/details/test/index.tsx]: I'm curious how changes related to transactions are connected to Deposit-related tests.

File: client/transactions/declarations.d.ts [hunk 2]
```diff
@@ -129,6 +129,12 @@ declare module '@woocommerce/navigation' {
 		type_is_not?: unknown;
 		source_device_is?: unknown;
 		source_device_is_not?: unknown;
+		channel_is?: unknown;
```
Comment (anu-rock) [inline client/transactions/declarations.d.ts]: Shouldn't the type for all the new fields be `string`? Perhaps our [TS guidelines](https://github.com/Automattic/woocommerce-payments/blob/develop/docs/typescript/README.md) provide hints?

File: client/transactions/list/index.tsx [hunk 3]
```diff
@@ -555,6 +561,14 @@ export const TransactionsList = (
 				!! search ||
 				!! typeIs ||
 				!! typeIsNot ||
+				!! typeIsNot ||
```
Comment (anu-rock) [inline client/transactions/list/index.tsx]: Possible duplication?

**Review by anu-rock — APPROVED** [review]
Pre-approving with minor comments. I could not do manual testing as my local setup is unexpectedly borked at the moment.

The only other feedback I have is that the new fields in the "add a filter" dropdown are center-aligned while the existing ones are left-aligned:

284641138-404cccaa-ee9d-4971-8f47-4fdb9beb91d2
___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ✅ All good |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | ✅ All good |
| Changelog  | ✅ All good |
| Testing instructions  | ✅ All good |
| Demo (video or JN site)  | ✅ All good |

File: client/transactions/list/index.tsx (hunk 3 above)
Comment (mordeth) [reply client/transactions/list/index.tsx]: Certainly, it's a duplication. It has been removed.

File: client/transactions/declarations.d.ts (hunk 2 above)
Comment (mordeth) [reply client/transactions/declarations.d.ts]: Well spotted! I have updated the types.

File: client/deposits/details/test/index.tsx (hunk 1 above)
Comment (mordeth) [reply client/deposits/details/test/index.tsx]: It appears that the code remains from investigating test failures. I believe we no longer require it.

**Comment by anu-rock** [comment]
Looks good to merge after the latest changes 🎉

**Pull request merged**